src/Oci8/Oci8Connection.php

```diff
@@ -215,75 +217,140 @@ protected function getDoctrineDriver()
      * Usage: DB::executeFunction('function_name(:binding_1,:binding_n)', [':binding_1' => 'hi', ':binding_n' =>
      * 'bye'], PDO::PARAM_LOB).
      *
-     * @author Tylerian - jairo.eog@outlook.com
-     * @param string $sql (mixed)
-     * @param array $bindings (kvp array)
-     * @param int $returnType (PDO::PARAM_*)
-     * @param int $length
+     * @param string $functionName
+     * @param array  $bindings   (kvp array)
+     * @param int    $returnType (PDO::PARAM_*)
+     * @param int    $length
      * @return mixed $returnType
      */
-    public function executeFunction($sql, array $bindings = [], $returnType = PDO::PARAM_STR, $length = null)
+    public function executeFunction($functionName, array $bindings = [], $returnType = PDO::PARAM_STR, $length = null)
     {
-        $query = $this->getPdo()->prepare('begin :result := ' . $sql . '; end;');
+        $stmt = $this->createStatementFromFunction($functionName, $bindings);
 
         foreach ($bindings as $key => &$value) {
-            if (! preg_match('/^:(.*)$/i', $key)) {
-                $key = ':' . $key;
-            }
-
-            $query->bindParam($key, $value);
+            $stmt->bindParam($key, $value);
         }
 
-        $query->bindParam(':result', $result, $returnType, $length);
-        $query->execute();
+        $stmt->bindParam(':result', $result, $returnType, $length);
+        $stmt->execute();
+
+        // auto convert strings like "343" and "23434.23" to int and float
+        $result = OracleTypeCaster::tryNumeric($result);
```

Comment:
Do we really need to cast the result? Maybe we can just let the developer handle the casting?